test/metabase/dashboard_subscription_test.clj

```diff
@@ -289,70 +289,71 @@
-                             {:type "section", :fields [{:type "mrkdwn", :text "Sent by Rasta Toucan"}]}]}
-                   {:title           pulse.test-util/card-name
-                    :rendered-info   {:attachments false, :content true, :render/text true},
-                    :title_link      (str "https://metabase.com/testmb/question/" card-id)
-                    :attachment-name "image.png"
-                    :channel-id      "FOO"
-                    :fallback        pulse.test-util/card-name}
-                   {:blocks [{:type "divider"}
-                             {:type "context"
-                              :elements [{:type "mrkdwn"
-                                          :text (str "<https://metabase.com/testmb/dashboard/"
-                                                     dashboard-id
-                                                     "?state=CA&state=NY&state=NJ&quarter_and_year=Q1-2021|*Sent from Metabase Test*>")}]}]}]}
-                 (pulse.test-util/thunk->boolean pulse-results)))))}}))
+  (with-redefs [metabase.pulse/attachment-text-length-limit 15]
+    (tests {:pulse     {:skip_if_empty false}
+            :dashboard pulse.test-util/test-dashboard}
+      "Dashboard subscription that includes a dashboard filters"
+      {:card (pulse.test-util/checkins-query-card {})
+
+       :fixture
+       (fn [_ thunk]
+         (mt/with-temporary-setting-values [site-name "Metabase Test"]
+           (thunk)))
+
+       :assert
+       {:email
+        (fn [_ _]
+          (testing "Markdown cards are included in email subscriptions"
+            (is (= (rasta-pulse-email {:body [{"Aviary KPIs" true
+                                               "<a class=\\\"title\\\" href=\\\"https://metabase.com/testmb/dashboard/\\d+\\?state=CA&amp;state=NY&amp;state=NJ&amp;quarter_and_year=Q1-2021\\\"" true}
+                                              pulse.test-util/png-attachment]})
+                   (mt/summarize-multipart-email #"Aviary KPIs"
+                                                 #"<a class=\"title\" href=\"https://metabase.com/testmb/dashboard/\d+\?state=CA&amp;state=NY&amp;state=NJ&amp;quarter_and_year=Q1-2021\"")))))
+
+        :slack
+        (fn [{:keys [card-id dashboard-id]} [pulse-results]]
+          (testing "Markdown cards are included in attachments list as :blocks sublists, and markdown is
+                  converted to mrkdwn (Slack markup language) and truncated appropriately"
+            (is (= {:channel-id "#general"
+                    :attachments
+                    [{:blocks [{:type "header", :text {:type "plain_text", :text "Aviary KPIs", :emoji true}}
+                               {:type "section",
+                                :fields [{:type "mrkdwn", :text (str (subs "*State*\nCA, NY and NJ" 0 14)       "…")}
+                                         {:type "mrkdwn", :text (str (subs "*Quarter and Year*\nQ1, 2021" 0 14) "…")}]}
```

Comment:
My personal bias is towards minimizing logic in the expected values of tests, so I'd probably just write these out as the literal values. Not a big deal either way though.